packages/headless/src/modules/store-admin-role/services/index.ts

```diff
@@ -0,0 +1,98 @@
+import { HttpStatus, Injectable } from '@nestjs/common';
+import { PERMISSIONS } from 'models';
+import { Role } from 'src/entity/role';
+import { StoreAdminInfo } from 'src/entity/store-admin';
+import { errorResponse, successResponse } from 'src/utils/response';
+import { IServiceResponse } from 'src/utils/response/service.response.interface';
+import { RoleRepository } from '../repositories';
+import { CreateRoleDto, UpdateRoleDto } from '../rest/dto/role.dto';
+
+@Injectable()
+export class RoleService {
+  constructor(private readonly roleRepository: RoleRepository) {}
+
+  async create(
+    adminInfo: StoreAdminInfo,
+    body: CreateRoleDto,
+  ): Promise<IServiceResponse<Role>> {
+    const isExist = await this.roleRepository.findOne({
+      name: body.name,
+      storeId: adminInfo.storeId,
+    });
+    if (isExist) {
+      return errorResponse(
+        'This role is already exist!',
+        null,
+        HttpStatus.CONFLICT,
+      );
+    }
+    const newRole = await this.roleRepository.create(body);
+    if (newRole) {
+      return successResponse(Role, newRole);
+    }
+    return errorResponse('Error in create new role', null, HttpStatus.CONFLICT);
+  }
+
+  async updateRole(
+    adminInfo: StoreAdminInfo,
+    body: UpdateRoleDto,
+  ): Promise<IServiceResponse<Role>> {
+    const isExist = await this.roleRepository.findOne({
+      id: body.id,
+      storeId: adminInfo.storeId,
+    });
+    if (isExist) {
+      if (isExist.name === body?.name) {
+        return errorResponse(
+          'This role is already exist!',
```

Comment:
nested branching is not good coding style. this creates clunky code. rather you could do something like
 `If (!isExist) {
        return ...
 }
other logic....
`

Reply:
Fixed